src/coreclr/gc/gc.cpp

```diff
@@ -38885,16 +39228,48 @@ void gc_heap::trim_youngest_desired_low_memory()
     }
 }
 
-ptrdiff_t gc_heap::estimate_gen_growth (int gen)
+ptrdiff_t gc_heap::estimate_gen_growth (int gen_number)
 {
-    dynamic_data* dd_gen = dynamic_data_of(gen);
-
+    dynamic_data* dd_gen = dynamic_data_of (gen_number);
+    generation *gen = generation_of (gen_number);
     ptrdiff_t new_allocation_gen = dd_new_allocation (dd_gen);
-    ptrdiff_t free_list_space_gen = generation_free_list_space (generation_of (gen));
+    ptrdiff_t free_list_space_gen = generation_free_list_space (gen);
+
+#ifdef USE_REGIONS
+    // in the case of regions, we assume all the space up to reserved gets used before we get a new region for this gen
+    ptrdiff_t reserved_not_in_use = 0;
+    ptrdiff_t allocated_gen = 0;
 
+    heap_segment* region = generation_tail_ro_region (gen);
```

Comment:
if there are ro regions, it means region would point to the last ro region instead of the 1st rw region. you can do

`heap_segment* region = heap_segment_rw (generation_start_segment (gen)); `

so you don't need to do the ro check yourself.

Reply:
Ah, yes, that's easier. But this would potentially go through a lot of ro segments. Should we add a helper on generation that gives back the first rw region, e.g. "generation_first_rw_segment" or similar?

Reply:
good point! yeh we should add a new helper like you said, probably `generation_start_segment_rw` is a better name to be more consistent with the existing `generation_start_segment`.